src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1015,6 +1015,57 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
                 throw new DaprException("State operation failed: the Dapr endpoint indicated a failure. See InnerException for details.", ex);
             }
         }
+
+        /// <inheritdoc/>
+        public async override Task<StateQueryResponse<TValue>> QueryStateAsync<TValue>(
+            string storeName,
+            string jsonQuery,
+            IReadOnlyDictionary<string, string> metadata = default,
+            CancellationToken cancellationToken = default)
+        {
+            var queryRequest = new Autogenerated.QueryStateRequest()
+            {
+                StoreName = storeName,
+                Query = jsonQuery
+            };
+
+            if (metadata != null)
+            {
+                foreach (var kvp in metadata)
+                {
+                    queryRequest.Metadata.Add(kvp.Key, kvp.Value);
+                }
+            }
+
+            var options = CreateCallOptions(headers: null, cancellationToken);
+
+            try
+            {
+                var items = new List<StateQueryItem<TValue>>();
+                var queryResponse = await client.QueryStateAlpha1Async(queryRequest, options);
+                foreach (var item in queryResponse.Results)
+                {
+                    if (!string.IsNullOrEmpty(item.Error))
+                    {
+                        // When we encounter an error, we throw an exception containing the results thus far.
+                        var partialResults = new StateQueryResponse<TValue>(items, string.Empty, queryResponse.Metadata);
+                        throw new StateQueryException<TValue>($"Encountered an error while processing state query results: {item.Error}", partialResults);
```

Comment:
You should do this in two passes. The problem with this is you get different data depending on order of the documents returned.

So first read the results and build `items`. Then loop again and check for errors. That way you always include all of the data.

Reply:
Done. I also included a list of the keys that failed so it's easier for a user to identify where things are going wrong/deal with the problematic keys.